msal/authority.py

```diff
@@ -53,11 +53,22 @@ def __init__(self, authority_url, http_client, validate_authority=True):
             performed.
         """
         self._http_client = http_client
-        authority, self.instance, tenant = canonicalize(authority_url)
+        self.authority_url = authority_url
+        self.validate_authority = validate_authority
+        self.is_initialized = False
+
+    def initialize(self):
+        if not self.is_initialized:
+            self.__initialize()
+            self.is_initialized = True
+
+    def __initialize(self):
```

Comment:
Looking better and better now. :-)  Perhaps we can try to go one step further, by doing this:

```suggestion
    def __initialize(self, authority_url, http_client, validate_authority):
```

so that its implementation would probably be exactly the same as the previous `__init__()`.